star-ros2/src/star_perception/star_perception/hailo_runner.py

```
@@ -0,0 +1,489 @@
+            "Rebuild the star_perception package.")
+    if not Path(python_bin).exists():
+        raise RuntimeError(
+            f"Hailo worker python {python_bin} not found. On Ubuntu 24.04 "
+            "create the venv per scripts/install_hailo_stack.sh.")
+
+    state = {"proc": None, "lock": threading.Lock()}
+
+    def _spawn() -> subprocess.Popen:
+        env = os.environ.copy()
+        env.pop("PYTHONPATH", None)  # don't leak ROS2's 3.12 paths into 3.11
+        proc = subprocess.Popen(
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)